### ISSUE TITLE
Don't look for opt/llc in $PATH

### ISSUE THREAD

**Issue by directhex** [mono/mono#9415]
The AOT compiler only works with our LLVM fork - e.g. passing Mono-specific flags in https://github.com/mono/mono/blob/master/mono/mini/aot-compiler.c#L9100

So requiring that users use pass `--aot=llvm,llvm-path=/Library/Frameworks/Mono.framework/Versions/Current/bin/` to the runtime is dumb. We know we only work with the llc/opt we built & bundled. We should know that path and use it, by default.

On Linux, I am patching this already: https://github.com/mono/linux-packaging-mono/blob/master/debian/patches/llvm_llc_opt_default_path.patch

Obviously this patch isn't usable as-is on every platform we support LLVM AOT, but that's my suggestion - use the right path by default, not `$PATH`.

**Comment by lewurm** [comment]
For what it's worth, as a developer we previously didn't run into this issue, because `./autogen.sh` required us to update `PATH` so that it points to the path wherever `llvm-config` lived. That coincidentally  also updated the `PATH` correctly for `opt` and `llc`.

**Comment by marek-safar** [comment]
@vargaz could you look into this

**Comment by vargaz** [comment]
Perhaps we should rename llc/opt to mono-llvm-llc/opt this will solve the issue of trying to invoke the stock llvm one.

**Comment by directhex** [comment]
@vargaz so put a mono-prefixed binary in $PATH, rather than an un-prefixed binary in some secret location?

**Comment by vargaz** [comment]
Yes. It won't solve the issue of finding the binaries, but will prevent using the wrong ones.

**Comment by directhex** [comment]
OK, yes, not using the wrong ones has value, but I think we should sill have a plan for finding the binaries somehow.

On Linux, if we prefix the binaries, they can just go in `/usr/bin` (same folder as mono-sgen itself), so that's easy. Right now we put `opt` in a private prefix, but that can change to align with any requirements.

On Mac, right now we put our `mono-sgen` in `Commands/` (which is in `$PATH`, and symlinks to), and our `opt` in `bin/` (which is not).

We aren't shipping LLVM on Windows yet, but want to - this is a problem case because Mono doesn't go into `$PATH` at all on Windows.

Could we try calling readlink on the mono process' own executable to work out where it lives, and try calling `opt` (or `mono-opt`, or whatever) from there? That _ought_ to solve all three cases, right?

Mac: 
`/Library/Frameworks/Mono.framework/Versions/Current/Commands/mono-sgen` -> `readlink()` -> `/Library/Frameworks/Mono.framework/Versions/Current/bin/mono-sgen` -> `dirname()` -> `/Library/Frameworks/Mono.framework/Versions/Current/bin/` and use that directory as the default to value of acfg->aot_opts.llvm_path

Linux:
`/usr/bin/mono-sgen` -> `readlink()` -> `/usr/bin/mono-sgen` -> `dirname()` -> `/usr/bin/` and use that directory as the default to value of acfg->aot_opts.llvm_path

Windows:
`C:\Program Files\Mono\bin\mono-sgen.exe` -> `readlink()` -> `C:\Program Files\Mono\bin\mono-sgen.exe` -> `dirname()` -> `C:\Program Files\Mono\bin\` and use that directory as the default to value of `acfg->aot_opts.llvm_path`

That ought to cover us for private prefixes not in `$PATH`, as is the Windows case.

The use case this would regress is where mono is in /usr/bin but llvm is in /usr/local/bin
